crates/ruff/src/cst/matchers.rs

```diff
@@ -67,7 +63,18 @@ pub(crate) fn match_aliases<'a, 'b>(
     }
 }
 
-pub(crate) fn match_call<'a, 'b>(expression: &'a mut Expression<'b>) -> Result<&'a mut Call<'b>> {
+#[allow(clippy::borrowed_box)]
+pub(crate) fn match_call<'a, 'b>(expression: &'a Expression<'b>) -> Result<&'a Box<Call<'b>>> {
```

Comment:
Can you explain why we want to borrow the box here instead of returning a `Result<&'a Call<'b>>`?

Reply:
The code does not compile without it because this code:
https://github.com/charliermarsh/ruff/blob/3b8121379ddec98d759c3ed21e3ed65bef122e97/crates/ruff/src/rules/flake8_comprehensions/fixes.rs#L706-L722

Reply:
I would have thought that there's a better way (feels very clumsy) but `Box::new((*inner_call).clone())` works.